api/jobworker/v1/jobworker.proto

```diff
@@ -0,0 +1,51 @@
+syntax = "proto3";
+
+package jobworker.v1;
+
+service JobWorkerService {
+  rpc StartJob(StartJobRequest) returns (StartJobResponse) {}
+  rpc StopJob(StopJobRequest) returns (StopJobResponse) {}
+  rpc JobStatus(JobStatusRequest) returns (JobStatusResponse) {}
+  rpc StreamJobOutput(StreamJobOutputRequest) returns (stream StreamJobOutputResponse) {}
+}
+
+message StartJobRequest {
+  string command = 1;
+  repeated string args = 2;
+}
+
+message StartJobResponse {
+  string job_id = 1;
+}
+
+message StopJobRequest {
+  string job_id = 1;
+}
+
+message StopJobResponse {
+  int32 exit_code = 1;
+}
+
+message JobStatusRequest {
+  string job_id = 1;
+}
+
+enum JobStatus {
+  JOB_STATUS_UNSPECIFIED = 0;
+  JOB_STATUS_RUNNING = 1;
+  JOB_STATUS_COMPLETE = 2;
+  JOB_STATUS_ERROR = 3;
+}
+
+message JobStatusResponse {
+  JobStatus status = 1;
+  int32 exit_code = 2;
```

Comment:
should we inform clients that exit_code is optional by making it a pointer to int32?

Reply:
good suggestion, updated